Read database configuration from profiles.clj, even for ragtime.

pupeno opened the pull request:
Related to: https://groups.google.com/forum/#!topic/luminusweb/rRJYbyUOKAY

After this change has been reviewed and/or accepted, I plan on working on a more seamless integration with Heroku and other places where the standard non-jdbc URLs are used as well as documenting the new way of using ragtime and configuring the database.

yogthos commented:
looks great!

yogthos commented:
I'll do a bit of local testing today and then I'm ok pushing it out. Going to have to review the docs on the site as well to make sure everything matches up. :)

yogthos commented:
@pupeno btw I updated to simply use `lein run migrate` and `lein run rollback`, this way it's not ragtime specific and less typing :)

yogthos commented:
I think source paths need to be added for the ragtime migrations directory to be packaged in the jar as well.

yogthos commented:
Also might make sense to update to `0.4.x` Ragtime as it looks a bit cleaner now.

yogthos commented:
it also now uses `resources` as its default path

pupeno commented:
I thought of doing lein run migrations and lein run rollback, but I wasn't sure if that was polluting the lein run *namespace*. I still intend on doing some improvements. I think in the long run, profiles.clj should be present regardless of whether you are using a database because it's the right place to put api keys and other secrets and passwords, so maybe we would need a flexible way for different luminus *components* or whatever you call the things beyond core) to inject profiles.clj. Ore maybe we should switch to profiles.d.

yogthos commented:
I wouldn't expect too many things ending up in lein run to be honest. The migrations is the first thing to come up so far really. Most of the tasks happen during development and can be done through plugins without having to resort to it.

I agree with the `profiles.clj` being present as the default. For the db stuff, ideally I'd like to have to work from the uberjar, that way you can push the app out to the server and have it apply migrations there.

It looks like ragtime 0.3.x branch expects migrations to live in the root of the project as opposed to under resources. I don't really want to add `resource-paths` just for it. Since 0.4.x defaults to `resources/migrations` that seems much cleaner to me.

If you're looking at doing more updates, perhaps you can take a look at moving to the latest ragtime. I also updated the mongo profile to use profiles.clj, so that should be in line with the rest now.

yogthos commented:
@pupeno there is another big problem here, the `:dev` in `profiles.clj` overwrites the `:dev` profile in `project.clj`. There needs to be a way to merge the `:dev` from profiles and project.

pupeno commented:
Ouch. I'll look into it as soon as I can. I'm putting out some
(metaphorical) fires right now.

On 25 June 2015 at 17:39, Dmitri Sotnikov <notifications@github.com> wrote:

> @pupeno <https://github.com/pupeno> there is another big problem here,
> the :dev in profiles.clj overwrites the :dev profile in project.clj.
> There needs to be a way to merge the :dev from profiles and project.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/luminus-framework/luminus-template/pull/129#issuecomment-115318522>
> .
>



-- 
J. Pablo Fernández <pupeno@pupeno.com> (http://pupeno.com)

yogthos commented:
I got the profile issue solved by changing it to `:provided` instead. I was looking a bit at ragtime `0.4.0` and there are some things I'm not too sure about regarding the new approach. It seems to be very much geared towards being used from within the REPL.

yogthos commented:
@pupeno so after doing a bit of research, I don't think ragtime provides the best workflow for this. I looked through all the available migration libs and the one I liked the best was actually Migratus. Unfortunately, it doesn't seem to be maintained any longer, so I ended up [forking it here](https://github.com/yogthos/migratus) and updating it to the latest version of clojure.java.jdbc and fixing some small annoyances. I'm going to see if Paul would be ok with the changes and if not I'll just release it from my fork instead.

pupeno commented:
I think I actually played with migratus some time ago. I think having a
straightforward way to run migrations and store credentials securely taking
into account production environments is very important, so your work is
very appreciated.

--
J. Pablo Fernández <pupeno@pupeno.com>
http://pupeno.com
On Jun 28, 2015 9:20 PM, "Dmitri Sotnikov" <notifications@github.com> wrote:

> @pupeno <https://github.com/pupeno> so after doing a bit of research, I
> don't think ragtime provides the best workflow for this. I looked through
> all the available migration libs and the one I liked the best was actually
> Migratus. Unfortunately, it doesn't seem to be maintained any longer, so I
> ended up forking it here <https://github.com/yogthos/migratus> and
> updating it to the latest version of clojure.java.jdbc and fixing some
> small annoyances. I'm going to see if Paul would be ok with the changes and
> if not I'll just release it from my fork instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/luminus-framework/luminus-template/pull/129#issuecomment-116325417>
> .
>

yogthos commented:
I just released a new version that uses Migratus, and I've updated the [docs](https://github.com/luminus-framework/luminus/blob/master/resources/md/migrations.md) to reflect it. From my testing everything looks good, but let me know if you run into any issues.

pupeno commented:
I just gave it a try and everything worked fine. I like how the code is
separated into namespaces now.

I'd like to have a new migration command, I guess this should be
implemented in the migratus lein plug in, right? lein migratus new or
something like that.

On 29 June 2015 at 17:36, Dmitri Sotnikov <notifications@github.com> wrote:

> I just released a new version that uses Migratus, and I've updated the
> docs
> <https://github.com/luminus-framework/luminus/blob/master/resources/md/migrations.md>
> to reflect it. From my testing everything looks good, but let me know if
> you run into any issues.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/luminus-framework/luminus-template/pull/129#issuecomment-116753134>
> .
>



-- 
J. Pablo Fernández <pupeno@pupeno.com> (http://pupeno.com)

pupeno commented:
I been trying to fix this https://github.com/yogthos/migratus-lein/issues/2 issue and the whole cli vs database implementations of migratus feel overly complex to the point that I'm not sure how this is solvable at all: https://github.com/yogthos/migratus-lein/issues/5

Since you seem to now be the official maintainer, shall we refactor it to be simpler and suit the Luminus needs?
